File: app/src/main/java/org/schabi/newpipe/util/DebounceSavable.java
package org.schabi.newpipe.util;

import org.schabi.newpipe.error.ErrorInfo;

public interface DebounceSavable {

    /**
     * Execute operations to save the data. <br>
     * Must set {@link DebounceSaver#setIsModified(boolean)} false in this method manually
     * after the data has been saved.
     */
    void saveImmediate();

    void showError(ErrorInfo errorInfo);
}

File: app/src/main/java/org/schabi/newpipe/util/debounce/DebounceSavable.kt
package org.schabi.newpipe.util.debounce

import org.schabi.newpipe.error.ErrorInfo

interface DebounceSavable {
    /**
     * Execute operations to save the data. <br></br>
     * Must set [DebounceSaver.setIsModified] false in this method manually
     * after the data has been saved.
     */
    fun saveImmediate()
    fun showError(errorInfo: ErrorInfo?)
}
